op-service/eth/blob_test.go

```diff
@@ -54,16 +94,20 @@ func TestInvalidBlobDecoding(t *testing.T) {
 	if err := b.FromData(data); err != nil {
 		t.Fatalf("failed to encode bytes: %v", err)
 	}
+
 	b[32] = 0x80 // field elements should never have their highest order bit set
-	if _, err := b.ToData(); err == nil {
-		t.Errorf("expected error, got none")
-	}
+	_, err := b.ToData()
+	require.Error(t, err)
+	b[32] = 0x0
 
-	b[32] = 0x00
-	b[4] = 0xFF // encode an invalid (much too long) length prefix
-	if _, err := b.ToData(); err == nil {
-		t.Errorf("expected error, got none")
-	}
+	b[VersionOffset] = 0x01 // invalid encoding version
+	_, err = b.ToData()
+	require.Error(t, err)
```

Comment:
Can we use the following method to assert specifically on certain errors? https://pkg.go.dev/github.com/stretchr/testify/require#ErrorIs

If we break out the errors into their own types then this becomes much easier. We should have test coverage of each error case

Reply:
done